### ISSUE TITLE
Adding a resource to a collection fails

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#5059]
**Description of the bug**
I tried to add a resource to a collection and received the following error
![image](https://user-images.githubusercontent.com/3456833/235454226-7946b26b-456f-4c3f-82ec-0a9f2023ee82.png)

Steps to reproduce the bug:
1. Go to a collection where you have edit permissions.  I was working on https://www.hydroshare.org/resource/b6c4fcad40c64c4cb4dd7d4a25d0db6e/ 
2. Switch to edit mode
3. Click on the button to Add resources
4. Select a Resource
5. The error occurs

**Expected behavior**
The resource should be added to the collection

**Additional information**
This may be related to #5057 as I had just prior to attempting to add, removed a resource from this collection.

**Comment by devincowan** [comment]
@dtarb do you remember what resource you were attempting to add to the collection when it failed? Thanks

**Comment by dtarb** [comment]
I was trying to add this resource https://www.hydroshare.org/resource/45b43d72928048a8bc10a009d932f769/.

**Comment by devincowan** [comment]
Tracking some info as I take a look
```
>>> resources = BaseResource.objects.filter(collections__short_id='b6c4fcad40c64c4cb4dd7d4a25d0db6e').all()
>>> resources
<QuerySet [<BaseResource: Great Salt Lake Area Volume Data>, <BaseResource: Great Salt Lake Basin Digital Elevation Model>, <BaseResource: Great Salt Lake Bathymetry>, <BaseResource: Great Salt Lake Mineral Extraction Water Withdrawals>, <BaseResource: Great Salt Lake Basin Watershed Boundaries Shapefile>]>
>>> resources.count()
5
>>> for res in resources:
...     print(res)
...     for rel in res.metadata.relations.all():
...         print(rel)
...
Great Salt Lake Area Volume Data
source Tarboton, D. (2017). Great Salt Lake Bathymetry, HydroShare, http://www.hydroshare.org/resource/582060f00f6b443bb26e896426d9f62a
isPartOf Tarboton, D. (2023). Collection of Great Salt Lake Data, HydroShare, http://www.hydroshare.org/resource/b6c4fcad40c64c4cb4dd7d4a25d0db6e
Great Salt Lake Basin Digital Elevation Model
isPartOf Tarboton, D. (2023). Collection of Great Salt Lake Data, HydroShare, http://www.hydroshare.org/resource/b6c4fcad40c64c4cb4dd7d4a25d0db6e
Great Salt Lake Bathymetry
references Baskin, R. L., (2005), "Calculation of Area and Volume for the South Part of Great Salt Lake, Utah," Open-File Report 2005–1327, USGS, Salt Lake City, http://pubs.water.usgs.gov/ofr20051327
isReferencedBy Tarboton, D. (2017). Great Salt Lake Area Volume Data, HydroShare, http://www.hydroshare.org/resource/89125e9a3af544eab2479b7a974100ba
references Baskin, R. L., (2006), "Calculation of Area and Volume for the North Part of Great Salt Lake, Utah," Open-File Report 2006–1359, USGS, Salt Lake City, http://pubs.water.usgs.gov/ofr20061359.
isPartOf Tarboton, D. (2023). Collection of Great Salt Lake Data, HydroShare, http://www.hydroshare.org/resource/b6c4fcad40c64c4cb4dd7d4a25d0db6e
Great Salt Lake Mineral Extraction Water Withdrawals
isDescribedBy Merck, M. and D. Tarboton, (2023 submitted), "The Salinity of the Great Salt Lake and its Deep Brine Layer," Water, https://www.mdpi.com/journal/water
isPartOf Tarboton, D. (2023). Collection of Great Salt Lake Data, HydroShare, http://www.hydroshare.org/resource/b6c4fcad40c64c4cb4dd7d4a25d0db6e
Great Salt Lake Basin Watershed Boundaries Shapefile
references Mohammed, I. N. and D. G. Tarboton, (2011), "On the Interaction between Bathymetry and Climate in the System Dynamics and Preferred Levels of the Great Salt Lake," Water Resour. Res., 47: W02525, http://dx.doi.org/10.1029/2010WR009561.
references Mohammed, I. N. and D. G. Tarboton, (2012), "An examination of the sensitivity of the Great Salt Lake to changes in inputs," Water Resour. Res., 48(11): W11511, http://dx.doi.org/10.1029/2012WR011908.
isPartOf Tarboton, D. (2023). Collection of Great Salt Lake Data, HydroShare, http://www.hydroshare.org/resource/b6c4fcad40c64c4cb4dd7d4a25d0db6e

**Comment by devincowan** [comment]
Here's something interesting
Taking a look at the resource that Dave attempted to add to the collection
```
>>> res = BaseResource.objects.filter(short_id='45b43d72928048a8bc10a009d932f769').all()
>>> rel = res.first().metadata.relations.all()
>>> for re in rel:
...     print(re)
...
isPartOf Tarboton, D. (2023). Collection of Great Salt Lake Data, HydroShare, http://www.hydroshare.org/resource/b6c4fcad40c64c4cb4dd7d4a25d0db6e
isVersionOf Tarboton, D. (2017). Great Salt Lake Level and Volume, HydroShare, http://www.hydroshare.org/resource/b26090299ec947c692d4ee4651815579
references Tarboton, D. (2017). Great Salt Lake Area Volume Data, HydroShare, http://www.hydroshare.org/resource/89125e9a3af544eab2479b7a974100ba
```
We can see the the resource that Dave tried to add
https://www.hydroshare.org/resource/45b43d72928048a8bc10a009d932f769/
"thinks" that is is already `partOf` the collection

**Comment by devincowan** [comment]
@pkdash looks like you're right, this is probably a regression that causes duplicate listing of resources in our modal
Probably introduced by PR https://github.com/hydroshare/hydroshare/pull/5038 as part of work on https://github.com/hydroshare/hydroshare/issues/5030

**Comment by devincowan** [comment]
fixed in 2.5.2 hotfix

**Comment by devincowan** [comment]
thanks @dtarb I went ahead and added the resource to your collection
Your resource had some "broken" metadata that needed to be removed.
Pabitra is currently reviewing a management command I wrote to find similarly "broken" resources and remedy them.

**Comment by devincowan** [comment]
looks like there are ~36 instances of "dangling" `isPartOf` metadata on prod:

11692/14295:WEPPcloud model for Cedar Creek, Bull Run Watershed, OR, http://www.hydroshare.org/resource/8b7ef268c81a4e92b9f866431023233c has isPart meta.
Collection WEPPcloud hydrologic model runs for undisturbed forested watersheds in the Lake Tahoe Basin, CA/NV, http://www.hydroshare.org/resource/c7d9e3d2052448f296e6d2aa8070f277 is missing hasPart.
Removing dangling isPartOf relation...
11693/14295:WEPPcloud model for Fir Creek, Bull Run Watershed, OR, http://www.hydroshare.org/resource/3a96ca9c9f0d4019b5da19cd88fc194c has isPart meta.
Collection WEPPcloud hydrologic model runs for undisturbed forested watersheds in the Lake Tahoe Basin, CA/NV, http://www.hydroshare.org/resource/c7d9e3d2052448f296e6d2aa8070f277 is missing hasPart.
Removing dangling isPartOf relation...
11694/14295:WEPPcloud model for Little Sandy, Bull Run Watershed, OR, http://www.hydroshare.org/resource/30c1694ee6f645c488c1374a2afcc0ef has isPart meta.
Collection WEPPcloud hydrologic model runs for undisturbed forested watersheds in the Lake Tahoe Basin, CA/NV, http://www.hydroshare.org/resource/c7d9e3d2052448f296e6d2aa8070f277 is missing hasPart.
Removing dangling isPartOf relation...
11695/14295:WEPPcloud model for North Fork, Bull Run Watershed, OR, http://www.hydroshare.org/resource/ac0cf7902a384658a3648c4130810ac8 has isPart meta.
Collection WEPPcloud hydrologic model runs for undisturbed forested watersheds in the Lake Tahoe Basin, CA/NV, http://www.hydroshare.org/resource/c7d9e3d2052448f296e6d2aa8070f277 is missing hasPart.
Removing dangling isPartOf relation...
11696/14295:WEPPcloud model for South Fork, Bull Run Watershed, OR, http://www.hydroshare.org/resource/525c512ee899485baf2cede46ee24d6b has isPart meta.
Collection WEPPcloud hydrologic model runs for undisturbed forested watersheds in the Lake Tahoe Basin, CA/NV, http://www.hydroshare.org/resource/c7d9e3d2052448f296e6d2aa8070f277 is missing hasPart.
Removing dangling isPartOf relation...
11697/14295:WEPPcloud model for Upper Cedar River, Cedar River Watershed, WA, http://www.hydroshare.org/resource/592190aa103c474fac818b3d0c05db08 has isPart meta.
Collection WEPPcloud hydrologic model runs for undisturbed forested watersheds in the Lake Tahoe Basin, CA/NV, http://www.hydroshare.org/resource/c7d9e3d2052448f296e6d2aa8070f277 is missing hasPart.
Removing dangling isPartOf relation...
11698/14295:WEPPcloud model for Taylor Creek, Cedar River Watershed, WA, http://www.hydroshare.org/resource/722979e575b2405c92e2f3d6937a12d8 has isPart meta.
Collection WEPPcloud hydrologic model runs for undisturbed forested watersheds in the Lake Tahoe Basin, CA/NV, http://www.hydroshare.org/resource/c7d9e3d2052448f296e6d2aa8070f277 is missing hasPart.
Removing dangling isPartOf relation...
11699/14295:WEPPcloud model for Watershed 3, Mica Creek Experimental Watershed, ID, http://www.hydroshare.org/resource/8c7dc32a87bc4c4cbd04c05262875d04 has isPart meta.
Collection WEPPcloud hydrologic model runs for undisturbed forested watersheds in the Lake Tahoe Basin, CA/NV, http://www.hydroshare.org/resource/c7d9e3d2052448f296e6d2aa8070f277 is missing hasPart.
Removing dangling isPartOf relation...
11700/14295:WEPPcloud model for Watershed 6, Mica Creek Experimental Watershed, ID, http://www.hydroshare.org/resource/5758f9322b514671b870a3d339ef80c8 has isPart meta.
Collection WEPPcloud hydrologic model runs for undisturbed forested watersheds in the Lake Tahoe Basin, CA/NV, http://www.hydroshare.org/resource/c7d9e3d2052448f296e6d2aa8070f277 is missing hasPart.
Removing dangling isPartOf relation...
11704/14295:Logan River Observatory: Logan River at Main Street (Highway 89/91) Bridge Aquatic Site (LR_MainStreet_BA) Quality Controlled Data, http://www.hydroshare.org/resource/9947db258e33414b82bf16d8e0463f3a has isPart meta.
Collection Logan River Observatory Datasets, http://www.hydroshare.org/resource/def147fa06de4c67810586d45337b413 is missing hasPart.
Removing dangling isPartOf relation...
11717/14295:Implementation of  WRFHydro 5.x model using CyberGIS-Compute V2, http://www.hydroshare.org/resource/329ede31b88942c489aca3111b076446 has isPart meta.
Collection CyberGIS-Jupyter for Water (CJW) 2021-Q4 Release Notes, http://www.hydroshare.org/resource/2086b241b16b453d827db847e8640475 is missing hasPart.
Removing dangling isPartOf relation...
11780/14295:MATLAB Online, http://www.hydroshare.org/resource/1c887cce1a72448c97892397613819a6 has isPart meta.
Collection CUAHSI compute services for working with data in the cloud, http://www.hydroshare.org/resource/aef4ecbc69374ba6b16851b9121878e1 is missing hasPart.
Removing dangling isPartOf relation...
11901/14295:Groundwater Hydrology/Hydrogeology Teaching Materials (full course), http://www.hydroshare.org/resource/327fae4ec11e4232b93a3c737bc05f7c has isPart meta.
Collection Educational Resources for Hydrology & Water Resources, http://www.hydroshare.org/resource/148b1ce4e308427ebf58379d48a17b91 is missing hasPart.
Removing dangling isPartOf relation...
11971/14295:Watershed boundaries, hydrography, stream gage locations, and impervious surface area shapefiles for Pennypack Creek watershed in Bucks, Montgomery and Philadelphia Counties, PA, USA, http://www.hydroshare.org/resource/33ab76d224a645be8b34817512c5556e has isPart meta.
Collection Dead Run Data Collection, http://www.hydroshare.org/resource/95cea97da29c44009730a5b942cedc2a is missing hasPart.
Removing dangling isPartOf relation...
11974/14295:Watershed boundaries, hydrography, stream gage locations, and impervious surface area shapefiles for NW Branch Anacostia watershed in Montgomery and Prince George's Counties, MD, USA, http://www.hydroshare.org/resource/dbf74d13c6e54558b2a48359f731639a has isPart meta.
Collection Dead Run Data Collection, http://www.hydroshare.org/resource/95cea97da29c44009730a5b942cedc2a is missing hasPart.
Removing dangling isPartOf relation...
11977/14295:Global Hydropower Simulation - 20th century, http://www.hydroshare.org/resource/7e3f7e70b7d84313a73407f651640afe has isPart meta.
Collection Global Hydropower Simulation Collection, http://www.hydroshare.org/resource/190d31948b6b48c8846683771a346cb5 is missing hasPart.
Removing dangling isPartOf relation...
12025/14295:Logan River Observatory: Spring Creek Storm Drain (LR_SC_SD) Raw Data, http://www.hydroshare.org/resource/99989b1f9d1d4d389cedd054d296c5b0 has isPart meta.
Collection Logan River Observatory Datasets, http://www.hydroshare.org/resource/def147fa06de4c67810586d45337b413 is missing hasPart.
Removing dangling isPartOf relation...
12120/14295:Waterhackweek 2019 Cyberseminar: Introduction to LandLab, http://www.hydroshare.org/resource/89140af3a43f483c8f142b5f6ad532fa has isPart meta.
Collection CUAHSI's 2019 Cyberseminar Series: Waterhackweek, http://www.hydroshare.org/resource/0b0113042df8426b84543b208c443c6b is missing hasPart.
Removing dangling isPartOf relation...
12146/14295:The Impact of Lateral Flow Parameterizations on Runoff of SUMMA Model at Reynolds Mountain East1, http://www.hydroshare.org/resource/c4f753df042548c086bdf7b824c9d51d has isPart meta.
Collection HS-1. Comparing Approaches to Achieve Reproducible Computational Modeling for Hydrological and Environmental Systems, http://www.hydroshare.org/resource/3a2686a69e6e4f07a85e4dcc4f017ba9 is missing hasPart.
Removing dangling isPartOf relation...
12160/14295:Wasatch Environmental Observatory Red Butte Network: Raw Data at Lower Knowlton Fork Aquatic (RB_LKF_A), http://www.hydroshare.org/resource/36d90220eaa24e789a3efb12dfcebe46 has isPart meta.
Collection IUTAH GAMUT Environmental Observatory Collected Datasets, http://www.hydroshare.org/resource/02b155615b794671bfc4c48870b3ce6f is missing hasPart.
Removing dangling isPartOf relation...
12173/14295:XTEST-Monie Bay - DTW, Water Temperature, Specific conductance - Mar 2021-ongoing, http://www.hydroshare.org/resource/c9076222efbe4acd84fecb2647ce1a53 has isPart meta.
Collection CCZN Monie Bay Data, http://www.hydroshare.org/resource/ea50226fcfba4461b958315305a32d7e is missing hasPart.
Removing dangling isPartOf relation...
12600/14295:IGUIDE Shapefile Testing Resource, http://www.hydroshare.org/resource/9d413b9d57824a79b8239a5f7c4fdf51 has isPart meta.
Collection Harvey Basemap Data Collections, http://www.hydroshare.org/resource/7661752c688a4f3ebcf58f8657773530 is missing hasPart.
Removing dangling isPartOf relation...
12674/14295:COPY Learning Module: CUAHSI JupyterHub, TimeSeries Data, http://www.hydroshare.org/resource/2443eb73e50e4a7fb514c8a70a042296 has isPart meta.
Collection CUAHSI Apps Learning Modules, http://www.hydroshare.org/resource/7c175dc6d043476cb022075d25b7dba1 is missing hasPart.
Removing dangling isPartOf relation...
12750/14295:Hydrology scenarios that characterize plausible future drought conditions in the Colorado River Basin, http://www.hydroshare.org/resource/4e04cd751e98409994970ca1a9b2c460 has isPart meta.
Collection Data Collection to Supplement the Future Hydrology of the Colorado River Basin Study, http://www.hydroshare.org/resource/6d351874f16947609eab585a81c3c60d is missing hasPart.
Removing dangling isPartOf relation...
12842/14295:How to use HydroShare to organize Summer Institute teamwork, http://www.hydroshare.org/resource/83b1a576dbac4e75acc81478137fb0ac has isPart meta.
Collection 2016 National Water Center Innovators Program on Flood Inundation and Disaster Response, http://www.hydroshare.org/resource/4a33d8abc8d54b58b6624eda42138d4a is missing hasPart.
Removing dangling isPartOf relation...
12926/14295:Milford Neck - Redox Potential - Mar 2022-ongoing, http://www.hydroshare.org/resource/a67d5e40ec0043b79cbc3e1fb7034578 has isPart meta.
Collection CCZN Monie Bay Data, http://www.hydroshare.org/resource/ea50226fcfba4461b958315305a32d7e is missing hasPart.
Removing dangling isPartOf relation...
13066/14295:Map of surface water frequency observed 2017-2020, Schaffer-Smith et al 2022, http://www.hydroshare.org/resource/47fdf2ea06ae49179fec93fdf18702fa has isPart meta.
Collection Tracking a blue wave of ephemeral water across arid southern Africa, http://www.hydroshare.org/resource/888b3a0520744f3283271a99d1fd5d8d is missing hasPart.
Removing dangling isPartOf relation...
13071/14295:Water quality synoptic data for Dead Run, Baltimore County, MD USA, http://www.hydroshare.org/resource/6af741cc432a4716ac797180c1ce3a2d has isPart meta.
Collection Dead Run Data Collection, http://www.hydroshare.org/resource/95cea97da29c44009730a5b942cedc2a is missing hasPart.
Removing dangling isPartOf relation...
13209/14295:Run WRFHydro Hands-on Training v5.2.x on CyberGIS-Jupyter for Water, http://www.hydroshare.org/resource/523fb35b0b0d4054a45e14aa8b742f78 has isPart meta.
Collection Collection of CyberGIS-Jupyter for Water (CJW) Example Notebooks, http://www.hydroshare.org/resource/0c5839097ff342118f30d6851a17071d is missing hasPart.
Removing dangling isPartOf relation...
13535/14295:test data, http://www.hydroshare.org/resource/e0172b52b8134bea806a626ff43c0656 has isPart meta.
Collection Test data collection, http://www.hydroshare.org/resource/9d458a50bfa54e6690a9c38b2ae066ac is missing hasPart.
Removing dangling isPartOf relation...
13536/14295:test data, http://www.hydroshare.org/resource/28e776f929e6460aadb0e9695e1494ba has isPart meta.
Collection Test data collection, http://www.hydroshare.org/resource/9d458a50bfa54e6690a9c38b2ae066ac is missing hasPart.
Removing dangling isPartOf relation...
13773/14295:Olympic Mountains Glacier Flowline Model, http://www.hydroshare.org/resource/d9d9a3dc020f4362901ddf9083c89551 has isPart meta.
Collection Glacier dynamics across precipitation gradients: Can ice-elevation feedbacks overcome the rain shadow in the Olympic Mountains, USA?, http://www.hydroshare.org/resource/a908fe06d4784684b68452d21efae69b is missing hasPart.
Removing dangling isPartOf relation...
13829/14295:Great Salt Lake Area Volume Data, http://www.hydroshare.org/resource/b884af46dc9d434bb4fdf3f2755a4401 has isPart meta.
Collection Collection of Great Salt Lake Data, http://www.hydroshare.org/resource/b6c4fcad40c64c4cb4dd7d4a25d0db6e is missing hasPart.
Removing dangling isPartOf relation...
14021/14295:Sauk-Suiattle HUC12 17110006, http://www.hydroshare.org/resource/0597a945890e4e5ab7ee9b8e9b1e43de has isPart meta.
Collection Sauk River Basin Observatory, http://www.hydroshare.org/resource/8f3d273a84d74ad1a1227a9f2b8d52ff is missing hasPart.
Removing dangling isPartOf relation...
14022/14295:Sauk-Suiattle HUC12 17110006, http://www.hydroshare.org/resource/ac5a2f6f7cef4c9e91a233f5588e2f38 has isPart meta.
Collection Sauk River Basin Observatory, http://www.hydroshare.org/resource/8f3d273a84d74ad1a1227a9f2b8d52ff is missing hasPart.
Removing dangling isPartOf relation...
14023/14295:Sauk-Suiattle HUC12 17110006, http://www.hydroshare.org/resource/f47ed9b5411a4282a966a46e2c278b73 has isPart meta.
Collection Sauk River Basin Observatory, http://www.hydroshare.org/resource/8f3d273a84d74ad1a1227a9f2b8d52ff is missing hasPart.
Removing dangling isPartOf relation...
36 isPart relations found dangling

**Comment by devincowan** [comment]
released in 2.5.3 hotfix today
